platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/TabStrip.kt

```diff
@@ -47,38 +59,124 @@ import org.jetbrains.jewel.ui.component.styling.TabStyle
  * @param style The visual styling configuration for the tab strip and its tabs
  * @param modifier Modifier to be applied to the tab strip container
  * @param enabled Controls the enabled state of the tab strip. When false, no tabs can be interacted with
+ * @param interactionSource The [MutableInteractionSource] that will be used to handle interactions with the tab strip
  * @see javax.swing.JTabbedPane
  * @see com.intellij.ui.components.JBTabbedPane
  */
 @Composable
-public fun TabStrip(tabs: List<TabData>, style: TabStyle, modifier: Modifier = Modifier, enabled: Boolean = true) {
+public fun TabStrip(
+    tabs: List<TabData>,
+    style: TabStyle,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+) {
     var tabStripState: TabStripState by remember { mutableStateOf(TabStripState.of(enabled = true)) }
+    val selectedIndex = remember(tabs) { tabs.indexOfFirst { it.selected } }
+    val tabCount = tabs.size
 
     remember(enabled) { tabStripState = tabStripState.copy(enabled) }
 
     val scrollState = rememberScrollState()
+
     Box(
-        modifier.focusable(true, remember { MutableInteractionSource() }).onHover {
-            tabStripState = tabStripState.copy(hovered = it)
-        }
+        modifier
+            .onFocusEvent { tabStripState = tabStripState.copy(focused = it.isFocused) }
+            .onHover { tabStripState = tabStripState.copy(hovered = it) }
+            .onPreviewKeyEvent { event ->
+                if (!enabled || tabs.isEmpty()) return@onPreviewKeyEvent false
+
+                when (event.type) {
+                    KeyEventType.KeyDown ->
+                        when (event.key) {
+                            Key.DirectionLeft -> {
+                                tabs[if (selectedIndex > 0) (selectedIndex - 1) else tabs.lastIndex].onClick()
+                                true
+                            }
+
+                            Key.DirectionRight -> {
+                                tabs[(selectedIndex + 1) % tabCount].onClick()
+                                true
+                            }
+
+                            else -> false
+                        }
+
+                    KeyEventType.KeyUp ->
+                        when (event.key) {
+                            Key.MoveHome -> {
+                                tabs.first().onClick()
+                                true
+                            }
+
+                            Key.MoveEnd -> {
+                                tabs.last().onClick()
+                                true
+                            }
+
+                            Key.Delete -> {
```

Comment:
Is this how tabs work in Swing? Never noticed you could close tabs with delete. I do know cmd/ctrl-w should close the current tab tho, at least for editor tabs (but we can let users do that if they want, I reckon)

Reply:
TBH I still need to implement a sample with Swing. This logic I got from the accessibility specs at https://www.w3.org/WAI/ARIA/apg/patterns/tabs/